deeppavlov/core/models/tf_model.py

```diff
@@ -13,119 +13,123 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
-
 import sys
 from abc import abstractmethod
+from collections import defaultdict
+import numpy as np
 
 import tensorflow as tf
-from overrides import overrides
 
 from deeppavlov.core.models.nn_model import NNModel
-from deeppavlov.core.models.serializable import Serializable
-from deeppavlov.core.common.attributes import check_attr_true
 from deeppavlov.core.common.log import get_logger
-from deeppavlov.core.common.errors import ConfigError
 from .tf_backend import TfModelMeta
 
-"""
-Here is an abstract class for neural network models based on Tensorflow.
-If you use something different, ex. Pytorch, then write similar to this class, inherit it from
-Trainable and Inferable interfaces and make a pull-request to deeppavlov.
-"""
-
 log = get_logger(__name__)
 
 
-class TFModel(Serializable, metaclass=TfModelMeta):
-    def __init__(self, **kwargs):
-        self._saver = tf.train.Saver
-        super().__init__(**kwargs)
+class TFModel(NNModel, metaclass=TfModelMeta):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
     @abstractmethod
-    def _add_placeholders(self):
-        """
-        Add all needed placeholders for a computational graph.
-        """
-        pass
+    def train_on_batch(self, x_batch, y_batch):
```

Comment:
This abstract method is defined in the parent class NNModel, so it's not needed here.

Reply:
The same thing with __call__, am I right?

Reply:
Yes